app/metal-controller-manager/controllers/serverclass_controller.go

```diff
@@ -30,127 +30,32 @@ type ServerClassReconciler struct {
-	for _, server := range sl.Items {
-		if server.Spec.Accepted {
-			newSF.items[server.Name] = server
-		}
-	}
-
-	return newSF
-}
-
-func (sr *serverResults) filterCPU(filters []metalv1alpha1.CPUInformation) serverFilter {
-	if len(filters) == 0 {
-		return sr
-	}
-
-	for _, server := range sr.items {
-		var match bool
-
-		for _, cpu := range filters {
-			if server.Spec.CPU != nil && cpu.PartialEqual(server.Spec.CPU) {
-				match = true
-
-				break
-			}
-		}
-
-		if !match {
-			// Remove from results list if it's there since it's not a match for this qualifier
-			delete(sr.items, server.ObjectMeta.Name)
-		}
-	}
-
-	return sr
-}
-
-func (sr *serverResults) filterSysInfo(filters []metalv1alpha1.SystemInformation) serverFilter {
-	if len(filters) == 0 {
-		return sr
-	}
-
-	for _, server := range sr.items {
-		var match bool
-
-		for _, sysInfo := range filters {
-			if server.Spec.SystemInformation != nil && sysInfo.PartialEqual(server.Spec.SystemInformation) {
-				match = true
-				break
-			}
-		}
-
-		if !match {
-			// Remove from results list if it's there since it's not a match for this qualifier
-			delete(sr.items, server.ObjectMeta.Name)
-		}
-	}
-
-	return sr
-}
-
-func (sr *serverResults) filterLabels(filters []map[string]string) serverFilter {
-	if len(filters) == 0 {
-		return sr
-	}
-
-	for _, server := range sr.items {
-		var match bool
-
-		for _, label := range filters {
-			for labelKey, labelVal := range label {
-				if val, ok := server.ObjectMeta.Labels[labelKey]; ok {
-					if labelVal == val {
-						match = true
-						break
-					}
-				}
-			}
-		}
-
-		if !match {
-			// Remove from results list if it's there since it's not a match for this qualifier
-			delete(sr.items, server.ObjectMeta.Name)
-		}
-	}
-
-	return sr
-}
-
-func (sr *serverResults) fetchItems() map[string]metalv1alpha1.Server {
-	return sr.items
-}
-
 // +kubebuilder:rbac:groups=metal.sidero.dev,resources=serverclasses,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=metal.sidero.dev,resources=serverclasses/status,verbs=get;update;patch
 // +kubebuilder:rbac:groups=metal.sidero.dev,resources=servers,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=metal.sidero.dev,resources=servers/status,verbs=get;update;patch
 
 func (r *ServerClassReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	ctx := context.Background()
+
 	l := r.Log.WithValues("serverclass", req.NamespacedName)
+	l.Info("reconciling")
+
+	//nolint:godox
+	// TODO: We probably should use admission webhooks instead (or in additional) to prevent
+	// unwanted edits instead of "fixing" the resource after the fact.
+	if req.Name == metalv1alpha1.ServerClassAny {
+		if err := ReconcileServerClassAny(ctx, r.Client); err != nil {
+			return ctrl.Result{}, err
+		}
 
-	l.Info("fetching serverclass", "serverclass", req.NamespacedName)
+		// do not return; re-reconcile it to update status
+	} //nolint:wsl
```

Comment:
The code above looks like a typical Go code with `return` at the end of the block. The comment explains why `return` is not there, that it is not forgotten. `wsl` linter's warning is… unfortunate.